client/http/encoding/json/json.go

```diff
@@ -0,0 +1,101 @@
+// Package json provides helper functions to handle requests and responses.
+package json
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github.com/beatlabs/patron/encoding"
+	"github.com/beatlabs/patron/encoding/json"
+	"github.com/beatlabs/patron/log"
+)
+
+// NewRequest creates a request and encodes the body and set's the appropriate headers.
+func NewRequest(ctx context.Context, method string, url string, payload interface{}) (*http.Request, error) {
+	buf, err := json.Encode(payload)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, method, url, bytes.NewBuffer(buf))
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set(encoding.ContentTypeHeader, json.Type)
+	req.Header.Set(encoding.ContentLengthHeader, strconv.FormatInt(int64(len(buf)), 10))
+	req.Header.Set(encoding.AcceptHeader, json.Type)
+
+	return req, nil
+}
+
+// FromResponse checks for valid headers and decodes th payload.
```

Comment:
Small typo suggestion here:
```suggestion
// FromResponse checks for valid headers and decodes the payload.
```